source/configuration_manual/mail_location/obox/amazon_s3.rst

```diff
@@ -35,35 +42,100 @@ partitions = this would theoretically provide you up to: ``~896,000
 
+Configuration
+^^^^^^^^^^^^^
+
 .. code-block:: none
 
   plugin {
-    obox_fs = fscache:1G:/var/cache/mails:compress:gz:6:s3: https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
-    obox_index_fs = compress:gz:6:s3: https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
+    obox_fs = fscache:1G:/var/cache/mails:compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
+    obox_index_fs = compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
   }
 
+.. Note::
+        If the version you are using does not support the ``aws-s3`` scheme,
+        try using the more generic ``s3`` scheme.
+
+        .. versionadded:: 2.3.10
+
 Get ACCESSKEY and SECRET from `www.aws.amazon.com <https://aws.amazon.com/>`_
 ``-> My account -> Security credentials -> Access`` credentials. Create the
 ``BUCKETNAME`` from ``AWS Management Console -> S3 -> Create Bucket``.
 
 If the ``ACCESSKEY`` or ``SECRET`` contains any special characters, they can be
 %hex-encoded.
 
+If the ``aws-s3`` scheme is used Dovecot defaults to prepend the following URL
+parameters, refer to :ref:`http_based_object_storages` for details. The same
+setup can be achieved by using the ``s3`` scheme and adding the parameters
+manually.
+
+.. code-block:: none
+
+  addhdrvar=x-amz-security-token:%{auth:token}&loghdr=x-amz-request-id&loghdr=x-amz-id-2
+
 .. Note::
 
   dovecot.conf handles %variable expansion internally as well, so % needs to be
   escaped as %% and ':' needs to be escaped as %%3A.
 
+
+AWS Signing v2 vs v4
+""""""""""""""""""""
+
 S3 driver uses the ``AWS2`` signing method by default, but ``AWS4`` can be used
 by adding the bucket region parameter to the S3 URL:
 
 .. code-block:: none
 
   plugin {
-    obox_index_fs = compress:gz:6:s3: https://ACCESSKEY:SECRET@host/?bucket=BUCKETNAME&region=eu-central-1
+    obox_index_fs = compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@host/?bucket=BUCKETNAME&region=eu-central-1
+  }
+
+Configuration using AWS IAM
+"""""""""""""""""""""""""""
+
+Dovecot supports AWS Identity and Access Management (IAM) for authenticating
+requests to AWS S3 using the AWS EC2 Instance Metadata Service (IMDSv2).
+
+Using IAM allows running Dovecot with S3 Storage while not keeping the
+credentials in the configuration.
+
+A requirement for using IMDSv2 is that Dovecot is running on an AWS EC2
+instance, otherwise the IMDS will not be reachable. Additionally an IAM role
+must be configured which allows trusted entities, EC2 in this case, to
+assume that role. The role (for example ``s3access``) that will be assumed must
+have the ``AmazonS3FullAccess`` policy attached.
+
+The ``auth_role`` can be configured as a URL parameter it specifies the IAM role
+to be assumed. If no ``auth_role`` is configured, no IAM lookup will be done.
+The IAM requests are targeted by default to the IMDSv2 service in EC2
+(``169.254.169.254``). It is possible to override this default by appending
+``auth_host=NEWHOST`` to the URL.
```

Comment:
I guess could mention also auth_port. And say that those are mainly useful for testing.